source/postprocess/visualization/velocity_residual.cc

```diff
@@ -0,0 +1,169 @@
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+#include <aspect/simulator.h>
+#include "velocity_residual.h"
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      VelocityResidual<dim>::
+      VelocityResidual ()
+        :
+        DataPostprocessorVector<dim> ("velocity_residual",
+        			update_values | update_quadrature_points | update_gradients)
+      {}
+
+
+      template <int dim>
+      void
+	  VelocityResidual<dim>::initialize ()
+	  {
+	    std::set<types::boundary_id> surface_boundary_set;
+	    surface_boundary_set.insert(this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"));
+	// The input ascii table contains one data column (velocity components) in addition to the coordinate columns.
+	    Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set, dim);
+	  }
+
+
+      template <int dim>
+      void
+      VelocityResidual<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+    	Assert ((computed_quantities[0].size() == dim), ExcInternalError());
+        auto cell = input_data.template get_cell<DoFHandler<dim> >();
+
+        const double velocity_scaling_factor =
+                      this->convert_output_to_years() ? year_in_seconds : 1.0;
+
+        for (unsigned int q=0; q<computed_quantities.size(); ++q)
+          for (unsigned int d = 0; d < dim; ++d)
+        	computed_quantities[q](d)= 0;
+
+        // We only want to output dynamic topography at the top and bottom
+        // boundary, so only compute it if the current cell has
+        // a face at the top or bottom boundary.
+        bool cell_at_top_boundary = false;
+        for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+          if (cell->at_boundary(f) &&
+              (this->get_geometry_model().translate_id_to_symbol_name (cell->face(f)->boundary_id()) == "top"))
+            cell_at_top_boundary = true;
+
+
+        if (cell_at_top_boundary)
+		{
+		  for (unsigned int q=0; q<computed_quantities.size(); ++q)
+		  {
+			Tensor<1,dim> velocity;
+		    if (use_spherical_unit_vectors)
+			{
+		      for (unsigned int d = 0; d < dim; ++d)
+		    	velocity[d] = Utilities::AsciiDataBoundary<dim>::get_data_component(
+		    	  							this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"),
+		    	  							input_data.evaluation_points[q], d);
+		      velocity = Utilities::Coordinates::spherical_to_cartesian_vector(velocity, input_data.evaluation_points[q]);
```

Comment:
We talked about the remaining rotations between the velocities. This line is the most likely point where this could be introduced. You will want to check if `velocity` that you read from the data file has the same order of columns  as this function expects (v_phi, v_theta), and if latitude and longitude are defined the same way (spherical vs geographical coordinates).

Reply:
Thank you for the suggestion! The problem was in fact with reading the vel_north (GPS data) as vel_theta and not -vel_theta.